### ISSUE TITLE
Cannot run container microsoft/iis anymore

### ISSUE THREAD

**Issue by ddobric** [Microsoft/Virtualization-Documentation#356]
Please note, this was working. When trying to start container, getting following error:

> docker run -d -p 80:80 microsoft/iis ping -t localhost
> 11f1444ec3532836dbf1f2c493b01d18fc27b86001fac94e7c8a86a5026cce2a
> failed to create endpoint hopeful_volhard on network nat

When I try it next time:
failed to create endpoint tiny_davinci on network nat:
...
failed to create endpoint ecstatic_shaw
...

Please note, this was working.

**Comment by ddobric** [comment]
?

**Comment by PatrickLang** [comment]
Can you try the following?
1) Stop all containers
2) `get-netnatstaticmapping | remove-netnatstaticmapping`
3) Try running `docker run -p 80:80 microsoft/iis` again

@JMesser81 - can you add this back to https://msdn.microsoft.com/en-us/virtualization/windowscontainers/management/container_networking if its still needed?

**Comment by ddobric** [comment]
It didn't help. It was one mapping defined and removed.

Error response from daemon: failed to create endpoint **hungry_williams** on network nat: HNS failed with error : Failed to create endpoint.

**Comment by JMesser81** [comment]
There does not appear to be a latent port mapping which isn't getting cleaned up based on @ddobric's response. Was this working on the same container host (VM)? What changed?

What is the IP address of the container host compared with the default NAT internal prefix (172.16.0.0/12). If they overlap there will be a problem.

**Comment by ezeeetm** [comment]
@ddobric do what @PatrickLang suggests, but reboot after step 2.

https://github.com/Microsoft/Virtualization-Documentation/issues/273

**Comment by alexellis** [comment]
I have the exact same issues.

I've tried removing the nat switch but it's made things worse. I've removed container + hyperv feature over and over again and nothing has been able to restore Docker to a working state.

Can someone please make a suggestion for resetting this to a working state?

```
PS C:\WINDOWS\system32> docker run -d --net=nat -p 1433:1433 --name sql --env sa_password=7?107>uoVzejpVeIN39 microsoft/
mssql-server-2016-express-windows
b19e140737ff395211b17e9a426c1b2ed472152c0aa2ad16a9c3c725a038c16d
c:\program files\docker\docker.exe: Error response from daemon: failed to create endpoint sql on network nat: HNS failed
 with error : The file cannot be opened because it is in the process of being deleted..
PS C:\WINDOWS\system32>
PS C:\WINDOWS\system32> Stop-Service docker
PS C:\WINDOWS\system32> Get-ContainerNetwork | Remove-ContainerNetwork

Confirm
Remove-ContainerNetwork will remove the container network "".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): a
Remove-ContainerNetwork : Unspecified error
At line:1 char:24
+ Get-ContainerNetwork | Remove-ContainerNetwork
+                        ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Remove-ContainerNetwork], VirtualizationException
    + FullyQualifiedErrorId : OperationFailed,Microsoft.Containers.PowerShell.Cmdlets.RemoveContainerNetwork

PS C:\WINDOWS\system32>  Get-NetNat | Remove-NetNat
PS C:\WINDOWS\system32> Get-VMSwitch | Remove-VMSwitch
PS C:\WINDOWS\system32>
```

**Comment by erichexter** [comment]
After yesterdays update, I was able to load the IIS image from a windows 10
pro machine. It seems like it is working with the current releases of
docker, fully patched win 10 and the Ga server releases.

Eric Hexter

blog | http://Hex.LosTechies.com
info | http://www.linkedin.com/in/erichexter

**Comment by hmdhasani** [comment]
@erichexter which version solved your problem? I have the exact same problem and did not fixed on 12.5 or even 13.0.-rc4

**Comment by PatrickLang** [comment]
I'm going to close this issue out as the original report was on the technical preview.
@hmdhasani can you start a thread with the error you're hitting on https://social.msdn.microsoft.com/Forums/en-US/home?forum=windowscontainers ?

I think it would also be worth trying to match the version of the container to that of the host. eg: if `(Get-ComputerInfo).WindowsBuildLabEx` returns ...14393.576, try `docker run Microsoft/iis:windowsservercore-10.0.14393.576`
